functions/src/lobby.ts

```diff
@@ -195,6 +198,32 @@ function startPrompt(lobbyDocRef: firestore.DocumentReference<Lobby>) {
   });
 }
 
+function setRole(lobbyDocRef: firestore.DocumentReference<Lobby>) {
+  const numCatFish = 1;
+  const check: number[] = [];
+  let current = 0;
+  const privatePlayerCollection = getPrivatePlayerCollection(lobbyDocRef);
+
+  return db.runTransaction(async (transaction) => {
+    const lobbyDoc = await transaction.get(lobbyDocRef);
+
+    const lobbyData = lobbyDoc.data();
+
+    for (let i = 0; i < numCatFish; i++) {
+      current = Math.floor(Math.random() * (lobbyData.uids.length - 1) + 0);
+      if (check.indexOf(current) === -1) {
+        check.push(current);
+      } else i--;
+    }
+
+    for (let j = 0; j < lobbyData.uids.length; j++) {
+      const privatePlayerDocRef = privatePlayerCollection.doc(lobbyData.uids[j]);
+
+      transaction.set(privatePlayerDocRef, { role: check.indexOf(j) === -1 ? "CAT" : "CATFISH" }, { merge: true });
+    }
+  });
+}
+
```

Comment:
this code is really hard to read. I can barly figure out what is going on maybe consider more desriptive variable names